src/pages/ExamsPage/hooks.js

```diff
@@ -137,3 +138,16 @@ export const useAllowancesData = () => {
   const allowancesList = useSelector(selectors.courseAllowancesList);
   return { allowancesList };
 };
+
+export const useCreateAllowance = () => {
+  const makeNetworkRequest = reduxHooks.useMakeNetworkRequest();
+  const courseId = useSelector(selectors.courseId);
+  return (formData) => {
+    const allowanceData = createAllowanceData(formData);
+    makeNetworkRequest({
+      requestKey: RequestKeys.createAllowance,
+      promise: api.createAllowance(courseId, allowanceData),
+      onSuccess: () => module.useFetchAllowances(),
+    });
+  };
+};
```

Comment:
If the idea is that after creating an allowance the allowance list gets updated, I think this won't work, because `useFetchAllowances()` is used similar to this hook. It returns a method to be used to fetch this data.

If I'm not mistaken, what we would need to do here is to initialize the hook at the start of the method, and then use it on success, something like:
```suggestion
export const useCreateAllowance = () => {
  const makeNetworkRequest = reduxHooks.useMakeNetworkRequest();
  const fetchAllowances = module.useFetchAllowances()
  const courseId = useSelector(selectors.courseId);
  
  return (formData) => {
    const allowanceData = createAllowanceData(formData);
    makeNetworkRequest({
      requestKey: RequestKeys.createAllowance,
      promise: api.createAllowance(courseId, allowanceData),
      onSuccess: () => fetchAllowances(courseId),
    });
  };
};
```

Reply:
Yes, I was just testing this locally and realized I need this change. Thank you!